File: Leetcode/Easy/Remove Linked List Elements.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* removeElements(ListNode* head, int val) {
        if(!head)return head;
        while(head && head->val==val)head=head->next;
        if(!head || !head->next)return head;
        ListNode *prev=head,*curr=head;
        while(curr && curr->next){
            curr=curr->next;
            while(curr && curr->val==val)curr=curr->next;
            prev->next=curr;
            prev=curr;
        }
        return head;
    }
};

File: Leetcode/Easy/Remove Linked List Elements.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode removeElements(ListNode head, int val) {
        ListNode ans = new ListNode();
        ListNode tmp = ans;
        while(head!=null){
            if(head.val!=val){
                tmp.next = new ListNode(head.val);
                tmp= tmp.next;
            }
            head= head.next;
        }
        return ans.next;
    }
}
